### PR TITLE
Auth devise

### PR DESCRIPTION
**Pull request by aliatayee**
## Add devise auth
In this milestone I have developed the following requirements:
### Project requirements
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,12 +1,22 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (mmsesay) [inline app/models/user.rb:4]: - [ ] It is required that the application asks for confirmation of email. In that case, please add devise's `:confirmable` in the user model to enable that feature.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,13 +6,33 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
     <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
```
Comment (mmsesay) [inline app/views/layouts/application.html.erb:8]: - [ ] The screenshot below shows the error I encountered when I tried testing the application. I think it has to do with tailwindcss configuration. I will suggest you look into it at your earliest convenience.

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @aliatayee, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - Good Readme :heavy_check_mark: 
 - No linter error :heavy_check_mark: 
 - Good PR title :heavy_check_mark: 
 - Good PR summary :heavy_check_mark:  


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (mmsesay) [reply app/views/layouts/application.html.erb:8]: | Error Screenshot |
:------------------------:|
| Screen Shot 2022-09-30 at 4 27 15 PM|

Comment (aliatayee) [reply app/views/layouts/application.html.erb:8]: Thank you for providing the feedback.
please follow the below setup process to make everything works fine
### Install & Setup

To setup and install this sample AlFa Blog project, follow the below steps:
- Clone this project by the command: 
  ```
  git clone https://github.com/aliatayee/AlFaBlog
  ```

- Then switch to the project folder by the bellow query:

  ```
  cd AlFaBlog
  ```

- Install the required dependecies with:
  ```
  bundle install
  ```
- Install npm dependencies with: 
  ```
  npm install
  ```
- Before running below commands update [config/database.yml](./config/database.yml) with your local `DB`credentials. Then run:
    ```
    rake db:create
    ```
    ```
    rake db:migrate
    ```
    ```
    rake db:seed
    ```
- To make the tailwindcss works run the below command:
    ```
    bundle exec rake assets:precompile
    ```

- With Ruby on Rails and PostgreSQL installed in your environment, you can run `rails server` to run the website.

**Comment by aliatayee** [comment]
please follow the below setup process to make everything works fine
### Install & Setup

To setup and install this sample AlFa Blog project, follow the below steps:
- Clone this project by the command: 
  ```
  git clone https://github.com/aliatayee/AlFaBlog
  ```

- Then switch to the project folder by the bellow query:

  ```
  cd AlFaBlog
  ```

- Install the required dependecies with:
  ```
  bundle install
  ```
- Install npm dependencies with: 
  ```
  npm install
  ```
- Before running below commands update [config/database.yml](./config/database.yml) with your local `DB`credentials. Then run:
    ```
    rake db:create
    ```
    ```
    rake db:migrate
    ```
    ```
    rake db:seed
    ```
- To make the tailwindcss works run the below command:
    ```
    bundle exec rake assets:precompile
    ```

- With Ruby on Rails and PostgreSQL installed in your environment, you can run `rails server` to run the website.

**Comment by mmsesay** [comment]
Thanks for sharing the installation guide. I think 'bundle exec rake assets:precompile' was missing.

Please request a re-review and I believe the next reviewer will try to follow these instructions in order to get your app working smoothly on their system.

**Review by uwadonat — APPROVED** [review]
## Hi @aliatayee,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Good pull request message ✔️
- Good communication with reviewer  ✔️

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**